Refactoring

marl1m opened the pull request:
Because this is the first contact the reviewer will have with your repository, it might be useful for the reviewer so see all your code.

Commit:
Initial commit

Commit:
Create ci.yml

Commit:
Update ci.yml

Commit:
Update pyproject.toml

Commit:
Create youmama.py

Commit:
remove ignores

Commit:
Update ci.yml

Commit:
refactoring the ci.yml

Commit:
Update README.md

marl1m committed:
Merge pull request #2 from marl1m/ci

Ci branch fully created

marl1m committed:
creating a branch named refactoring

 expandanded original clean_data() function into 3 different purpose functions

marl1m committed:
cleaning.py and tests refactored

marl1m committed:
Update pyproject.toml

marl1m committed:
update on code styling

marl1m committed:
Update test_cleaning.py

marl1m committed:
UPDATED CONSTANTS

marl1m committed:
bug fixing

marl1m committed:
bug fixing

marl1m committed:
bug fixing

marl1m committed:
changing constants

marl1m committed:
trying new ways to make it work

marl1m committed:
tests passed

marl1m committed:
new test added for the main() function

marl1m committed:
update on test main

marl1m committed:
Update test_cleaning.py

marl1m committed:
Update cleaning.py

marl1m committed:
refactoring of main and test main

marl1m committed:
--cov of 97%

marl1m committed:
update on test_main with --cov of 97%

marl1m committed:
Update test_cleaning.py

marl1m committed:
linting updated

marl1m committed:
Merge branch 'empty_branch' into refactoring

Nynor-code commented:
Boa Martim, O teu código está muito claro, facil de seguir, e faz tudo o que foi solicitado no assignment 2, parabéns!

Gostei do do facto de teres incluído um CLI simples com argparse – isso torna o código reutilizável tanto em scripts como em pipelines automatizados.

Excelente também a utilização do melt, eu não me tinha lembrado dessa forma.

LGTM 👍

Junto umas pequenas melhorias e sugestões:

**Código legóvel:**
* Como indiquei o código está muito claro, fixe!
* Ser consistente com type hints (country: str, output_path: Path | str, etc.).
* Evitar df_melted.year = ... e usar sempre df_melted["year"] = ....

**Comentários:**
* Tens docstrings curtas e diretas, o que é bom.
* Não sei se o uncle Bob gosta, mas tenho o habito de enriquecer as docstrings com informação dos parâmetros e retorno da função

**Nomes de variáveis**
* Os nomes são descritivos e consistentes (df_melted ou OUTPUT_DIR, etc)
* Pequeno detalhe: no fim usas data – talvez fosse mais claro chamar df_clean.
* Estopu certo que embora não estaje a ser usada a variável FIXTURES_DIR vai ser usada já de seguida ;-)

**Funções & Responsabilidade Única:**
* Cada função faz uma única tarefa e é curta – muito bem conseguido.

**Robustez:**
* uma sugestão de Copiar o DataFrame no início de clean_data para não alterar o original: 
```df = df.copy()```
* Validar a estrutura da primeira coluna antes de fazer o split.
* Usar .drop(columns=...) com o nome real da coluna em vez do índice.
* Este bloco parece-me redundante (a coluna country nunca existe no df original):
```
if "country" in df.columns:
    data = data[data["country"] == country]
```
* Garantir que o folder de output existe antes de gravar:
```
Path(output_path).parent.mkdir(parents=True, exist_ok=True)
```
* Vais precisar tratar o campo geo/time porque além dos países tem regiões tipo "EU27_2007, etc." neste momento so filtras por PT.
* Atenção aos campos vazios e ao seu tratamento. Não tenho a certeza de que temos vazios com geo == "PT".

Nynor-code reviewed (approved):
Boa Martim, O teu código está muito claro, facil de seguir, e faz tudo o que foi solicitado no assignment 2, parabéns!

Gostei do do facto de teres incluído um CLI simples com argparse – isso torna o código reutilizável tanto em scripts como em pipelines automatizados.

Excelente também a utilização do melt, eu não me tinha lembrado dessa forma.

LGTM 👍

**Junto umas pequenas melhorias e sugestões:**

**Código legóvel:**

- Como indiquei o código está muito claro, fixe!
- Ser consistente com type hints (country: str, output_path: Path | str, etc.).
- Evitar `df_melted.year = ...` e usar sempre `df_melted["year"] = ....`

**Comentários:**

- Tens docstrings curtas e diretas, o que é bom.
- Não sei se o uncle Bob gosta, mas tenho o habito de enriquecer as docstrings com informação dos parâmetros e retorno da função

**Nomes de variáveis**

- Os nomes são descritivos e consistentes (`df_melted` ou `OUTPUT_DIR`, etc)
- Pequeno detalhe: no fim usas data – talvez fosse mais claro chamar df_clean.
- Estopu certo que embora não estaje a ser usada a variável FIXTURES_DIR vai ser usada já de seguida ;-)

**Funções & Responsabilidade Única:**

- Cada função faz uma única tarefa e é curta – muito bem conseguido.

**Robustez:**

- uma sugestão de Copiar o DataFrame no início de clean_data para não alterar o original: `df = df.copy()`
- Validar a estrutura da primeira coluna antes de fazer o split.
- Usar `.drop(columns=...)` com o nome real da coluna em vez do índice.
- Este bloco parece-me redundante (a coluna country nunca existe no df original):
```
if "country" in df.columns:
    data = data[data["country"] == country]
```
- Garantir que o folder de output existe antes de gravar:
 `Path(output_path).parent.mkdir(parents=True, exist_ok=True)`
- Vais precisar tratar o campo geo/time porque além dos países tem regiões tipo "EU27_2007, etc." neste momento so precisas de filtrar por PT, pelo que não vai fazer falta.
- Atenção aos campos vazios e ao seu tratamento. Não tenho a certeza de que temos vazios com `geo == "PT"`.

Commit:
refactoring cleaning and its test

following some of the feedback from the reviewer and also starting to refactor the testing files

Commit:
update on the testing file

several new tests added

marl1m commented:
Obrigado pela review bastante detalhada, ja alterei algumas inconsistencias que encontraste (nomeadamente docstrings mais completas; alterei como sugeriste alguns nomes de variaveis para serem mais legiveis/percetiveis)
Ja adiantei alguns tests novos tambem do assignment 3 dai o PR agora nao ficar verdinho

Nynor-code reviewed (approved):
**Request changes** - após passar os testes. Ainda estão a falhar alguns itens.

A base do erro está em passares a fixture no decorator... penso que funciona melhor e parece-me mais simples se passares como argumento da função de teste:

```
def test_example(sample_data):  # sample_data é o fixture
    assert not sample_data.empty
    assert "col1" in sample_data.columns
```

Manténs o código simples, limpo e bem documentado 👍. Fixe!

Uma sugestão:
O pylint não está a avaliar os ficheiros de teste (tens um 'omit' para os ficheiros de teste no pyproject.toml) se o desativares vais encontrar algumas correções a fazer ao código dos testes.

Não sei se é prática, mas pessoalmente gosto de correr os testes nestes ficheiros também porque considero que também são código... ;-)

Commit:
updating conftest.py

marl1m committed:
updated conftest

Commit:
Update cleaning.py

Nynor-code reviewed (approved):
LGTM 👍

Pergunta: Não estás ainda a gravar o CSV para outros países, certo?
